### ISSUE TITLE
Suggestion: Enable installing a flatpak with a particular commit

### ISSUE THREAD

**Issue by ludvigng** [flatpak/flatpak#4154]
This is more of a suggestion than an issue, but currently there's no way of installing a particular commit rather than the latest. If you want to install an earlier commit you would have to install the latest and then downgrade with `flatpak update --commit= `.

Basically I'm asking if you could add the --commit to the install command too, streamlining this process a bit.

**Comment by kevinsmia1939** [comment]
I think you can already do it.
Like this.

    sources:
      - type: git
        url: https://github.com/OpenChemistry/tomvizsuper
        commit: 51f93b92edf9a65f858896a72a97aff184409b80

**Comment by RalfJung** [comment]
I would also very much appreciate a way to have single command that installs a flatpak at a particular commit. Sadly it is not that uncommon that updating a flatpak causes problems, and sometimes one has to experiment with uninstalling and reinstalling. Currently I have to `flatpak install` and then immediately `flatpak update --commit`, which is a bit unnecessarily indirect.

> I think you can already do it.

I have no idea what to do with this snippet of code, it doesn't look like a command that I can run in the shell. I am looking for `flatpak install --commit be96c3ecc66eb95ae045f7bd988f93f273d0fbc5e966ba38e166cb61f7aecf76 com.vscodium.codium`, or something with equivalent functionality.

**Comment by hfiguiere** [comment]
`flatpak update --help`

it will show a `--commit`.

To know the value of the commit try:

`flatpak remote-info --log` _origin_ _package_

**Comment by hfiguiere** [comment]
it doesn't exist for `flatpak install`

This is not really a standard way to use flatpak

**Comment by ludvigng** [comment]
True, it's not a standard way of using flatpak, but the behaviour itself is described in the official docs: https://docs.flatpak.org/en/latest/tips-and-tricks.html

This would basically just be a way of shortcutting the behaviour instead of having to first install a package and then do a downgrade (or the thing kevinsmia1939 described). But I understand if it's not a priority, since it's probably a very very rare use case. 🙂

**Comment by xMAC94x** [comment]
This is a blocker for https://github.com/veloren/veloren/ to be distributed by flatpak.
In case we ever want to produce a flatpak package it needs to be automatized, and flathub seems like a good solution for it.
However I cannot just define a specific version of our build-toolchain. I need to provide a package and flatpak (automatically, without asking) just installing the newest isn't working for us. we cannot just run update afterwards.
We need something like `flatpak install --commit`

And actually, for usability reasons, it would be awesome to provide something like `flatpak install --version`
for the average cli user.

Its super frustrating to run `flatpak install -v org.freedesktop.Sdk.Extension.rust-nightly/x86_64/23.08/1.78.0-nightly-20240115` just to see that the `1.78.0-nightly-20240115` got replaced by an `*` silently and you find out you installed the newest version

**Comment by maxborowsky** [comment]
[ludvigng](https://github.com/ludvigng)
> since it's probably a very very rare use case. 🙂

Not such a very rare case: for example, I use Ubuntu 22.04 LTS and want to install a specific version of Extension Manager that will work with GNOME Shell 42 - the latest version that meets this requirement is 0.3.2. And this is a newer version than the one offered in the Ubuntu repo - but only available with Flathub.

**Comment by oneEyedCharlie** [comment]
It's sad that here in 2025 we still can't do a flatpak install --commit=...  It's hard to convince people to take flatpak seriously when basic functionality like this is missing.

**Comment by PedroC88** [comment]
I would also argue this is a most... currently the Microsoft Edge installer is broken because the URL is broken on the "manifest?" (not sure how it's called). Probably the previous would install fine because the deb exists in the URL, but the new one is broken, (yesterday it was completely missing and today they pushed a new 145 but it's not the same build).

Since I'm on a fresh Linux install (after a decade away from Linux) I can't "update" to a commit because it's just not installed, and I can't install it because it's broken, and I can't choose the previous version.

To clarify, this scenario wouldn’t be the same as others previously mentioned of “automating the install and update”. This would be to execute the commit directly and ignore anything newer. If a package is broken you need to be able to install a previous working version.
